### ISSUE TITLE
Lint/Void should treat "case" and "if" the same way

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14756]
## Steps to reproduce the problem

```ruby
foo = 1

case foo when 1 then 2 end
puts 3

2 if foo == 1
puts 3
```

## Expected behavior

Both should get `Literal 2 used in void context. [Lint/Void]`

## Actual behavior

The second does. The first doesn't. 

## RuboCop version

```
System Version: macOS 26.1 (25B78)
Model Identifier: MacBookAir10,1
ruby 3.4.8 (2025-12-17 revision 995b59f666) +PRISM [arm64-darwin25]
rubocop     2025-12-22 edca36ebb

```

**Comment by forthrin** [comment]
`[1].each { next 2 }.then { pp it }` is another example of void content not getting pulled off the highway.

**Comment by forthrin** [comment]
Confirmed as working

**Comment by forthrin** [comment]
This seems to have introduced a false positive when a particular case should do nothing and otherwise something.

```ruby
case foo
when 'bar'
  nil # [Lint/Void]
else
  puts 'baz'
end
puts 'qux'

```

**Comment by bbatsov** [comment]
@forthrin Best to file a separate issue for this.
